test/index.js

```diff
@@ -155,4 +207,108 @@ describe('GoodLoggly', () => {
+    });
+
+    it('releases after maxDelay if threshold is not met', (done) => {
+      const stream = readStream();
+      const options = Merge({ threshold: 100, maxDelay: 150 }, logglyOptions, optionalFields);
+      const reporter = new GoodLoggly(options);
+
+      Loggly.Loggly.prototype.log = function (event, tags) {
+        expect(event).to.be.an.array();
+        expect(event).to.have.length(3);
+        expect(event[0]).to.equal(Merge({}, bulkResult[0], optionalFields));
+        expect(event[1]).to.equal(Merge({}, bulkResult[1], optionalFields));
+        expect(event[2]).to.equal(Merge({}, bulkResult[2], optionalFields));
+        expect(tags).to.be.a.function();
+        done();
+      };
+
+      stream.pipe(reporter);
+      stream.push(bulkLogEventData[0]);
+      stream.push(bulkLogEventData[1]);
+      setTimeout(() => {
+        stream.push(bulkLogEventData[2]);
+      }, 200);
+    });
+
+    it('clears the buffer on termination of stream', (done) => {
+      const stream = readStream();
+      const options = Merge({ threshold: 100 }, logglyOptions, optionalFields);
+      const reporter = new GoodLoggly(options);
+
+      Loggly.Loggly.prototype.log = function (event, tags) {
+        expect(event).to.be.an.array();
+        expect(event).to.have.length(3);
+        expect(event[0]).to.equal(Merge({}, bulkResult[0], optionalFields));
+        expect(event[1]).to.equal(Merge({}, bulkResult[1], optionalFields));
+        expect(event[2]).to.equal(Merge({}, bulkResult[2], optionalFields));
+        expect(tags).to.be.a.function();
+        done();
+      };
+
+      stream.pipe(reporter);
+      stream.push(bulkLogEventData[0]);
+      stream.push(bulkLogEventData[1]);
+      stream.push(bulkLogEventData[2]);
+      stream.push(null);
+    });
+
+    it('does nothing if buffer is empty on termination of stream', (done) => {
+      const stream = readStream();
+      const options = Merge({ threshold: 100 }, logglyOptions, optionalFields);
+      const reporter = new GoodLoggly(options);
+
+      Loggly.Loggly.prototype.log = function (event, tags) {
+        Code.fail('We should never get here');
+      };
+
+      stream.pipe(reporter);
+      stream.push(null);
+      done();
+    });
+
+    it('resets buffer after a send', (done) => {
+      const stream = readStream();
+      const options = Merge({ threshold: 2 }, logglyOptions, optionalFields);
+      const reporter = new GoodLoggly(options);
+
+      Loggly.Loggly.prototype.log = function (event, tags) {
+        expect(event).to.be.an.array();
+        expect(event).to.have.length(2);
+        expect(event[0]).to.equal(Merge({}, bulkResult[0], optionalFields));
+        expect(event[1]).to.equal(Merge({}, bulkResult[1], optionalFields));
+        expect(tags).to.be.a.function();
+        tags(null);
+      };
+
+      stream.pipe(reporter);
+      stream.push(bulkLogEventData[0]);
+      stream.push(bulkLogEventData[1]);
+      setTimeout(() => {
```

Comment:
This one needs setTimeout to trigger exceeding the `maxDelay`. The other below can change to immediate.